src/index.ts

```diff
@@ -33,18 +34,29 @@ class ServerlessDynaliteLocal {
     this.log = (str: string, pluginName: string) =>
       serverless.cli.log(str, pluginName);
     this.tables = getCreateTableCommandInput(serverless.service.resources);
+
+    if (serverless.service.custom.dynalite?.port != null) {
+      this.port = serverless.service.custom.dynalite.port;
+    }
+
+    if (serverless.service.custom.dynalite?.path != null) {
+      this.path = serverless.service.custom.dynalite.path;
+    }
+
+    if (serverless.service.custom.dynalite?.inMemory) {
+      this.path = undefined;
+    }
   }
 
   async startHandler() {
     const server = dynalite({ path: this.path });
-    await new Promise<void>((resolve, reject) =>
-      server.listen(this.port, (err: string | undefined) =>
-        err ? reject(err) : resolve()
-      )
+    await new Promise<void>((resolve) =>
+      server.listen(this.port, () => resolve())
```

Comment:
does this reject the promise if the server start fails? What was there before is my default "turn a callback into a promise" boilerplate.

Reply:
As far as I can see from the [node documentation](https://nodejs.org/docs/latest-v14.x/api/net.html#net_event_listening) and from testing, `listen` does not pass any parameters to the callback, so with the previous code, `reject` would never have been called.

If you want to handle errors, you instead have to call `server.on('error', err => ...)` before calling `listen`. But if you don't do this, the error is thrown as an unhandled exception instead, which I think is fine in this case, so therefore we don't need any error handler.